packages/lambda-powertools-middleware-stop-infinite-loop/index.js

```diff
@@ -0,0 +1,18 @@
+const CorrelationIds = require('@perform/lambda-powertools-correlation-ids')
+const Log = require('@perform/lambda-powertools-logger')
+
+module.exports = (threshold = 10) => {
+  return {
+    before: (handler, next) => {
+      const len = CorrelationIds.get()['call-chain-length'] || 1
+      if (len >= threshold) {
+        let awsRequestId = handler.context.awsRequestId
+        let invocationEvent = JSON.stringify(handler.event)
+        Log.error('Possible infinite recursion detected, invocation is stopped.', { awsRequestId, invocationEvent })
+        throw new Error(`'call-chain-length' reached threshold of ${threshold}, possible infinite recursion`)
```

Comment:
I'm not too sure about the best approach when using `middy`, but doesn't it prefer...

```suggestion
        next(new Error(`'call-chain-length' reached threshold of ${threshold}, possible infinite recursion`))
```

so other middlewares are called?

Reply:
I had a look at the `validator` middleware that's part of middy, and it was just throwing. I'm not sure what's the best approach here either, let me ask.

Reply:
so, apparently, there's no consensus on what's the best way to do this.